web/docs/project/testing.md

```diff
@@ -1,3 +1,171 @@
 ---
 title: Testing
 ---
+
+:::info
+Wasp is in beta, so keep in mind there might be some kinks / bugs, and possibly some changes with testing support in the future. If you encounter any issues, reach out to us on [Discord](https://discord.gg/rzdnErX) and we will make sure to help you out!
+:::
+
+## Testing Your React App
+
+Wasp enables you to quickly and easily write both unit tests and React component tests for your frontend code. Because Wasp uses [Vite](https://vitejs.dev/), we support testing web apps through [Vitest](https://vitest.dev/).
+
+<details>
+  <summary>Included Libraries</summary>
+  <div>
+
+  [`vitest`](https://www.npmjs.com/package/vitest): Unit test framework with native Vite support.
+
+  [`@vitest/ui`](https://www.npmjs.com/package/@vitest/ui): A nice UI for seeing your test results.
+
+  [`jsdom`](https://www.npmjs.com/package/jsdom): A web browser test environment for Node.js.
+
+  [`@testing-library/react`](https://www.npmjs.com/package/@testing-library/react) / [`@testing-library/jest-dom`](https://www.npmjs.com/package/@testing-library/jest-dom): Testing helpers.
+
+  [`msw`](https://www.npmjs.com/package/msw): A server mocking library.
+
+  </div>
+</details>
+
+### Setting Up Tests
+
+In order for Wasp to pick up your tests, they should be placed within the `src/client` directory and use an extension that matches [these glob patterns](https://vitest.dev/config#include). Some of the file names that Wasp will pick up as tests:
+
+- `yourFile.test.ts`
+- `YourComponent.spec.jsx`
+
+Within test files, you can import your other source files as usual. For example, if you have a component `Counter.jsx`, you test it by creating a file in the same directory called `Counter.test.jsx` and import the component with `'./Counter'`.
+
+### Running Tests
+
+Running `wasp test client` will start Vitest in watch mode and recompile your Wasp project when changes are made.
+
+- If you want to see a realtime UI, pass `--ui` as an option.
+- To run the tests just once, use `wasp test client run`.
+
+All arguments after `wasp test client` are passed directly to the Vitest CLI, so check out [their documentation](https://vitest.dev/guide/cli.html) for all of the options.
+
+:::warning
```

Comment:
<img width="609" alt="Screenshot 2023-07-25 at 11 16 08" src="https://github.com/wasp-lang/wasp/assets/2223680/bdbf2308-f39a-4e33-824d-ad01f5d57463">

Nit: You can write extra text after `warning` to change the title of the warning to something like `Be careful` which reads better IMHO than `Danger`

<img width="743" alt="Screenshot 2023-07-25 at 11 17 56" src="https://github.com/wasp-lang/wasp/assets/2223680/8f3d5bc0-0f43-4435-8862-b0679da4f263">